src/components/ConnectButton/WalletDropdownButton.tsx

```diff
@@ -0,0 +1,62 @@
+"use client";
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import { useAccount, useDisconnect } from "wagmi";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger
+} from "@/components/ui/dropdown-menu";
+import { Icons } from "../Icons";
+import NoSSR from "../NoSSR";
+import { Button } from "@/components/ui/button";
+import ConnectButton from "./ConnectButton";
+
+export default function WalletDropdownButton() {
+  const { address } = useAccount();
+  const { disconnect } = useDisconnect();
+  const { push } = useRouter();
+  const [isCopied, setIsCopied] = useState(false);
+
+  const onCopyAddress = () => {
+    navigator.clipboard.writeText(address as string);
+    setIsCopied(true);
+    setTimeout(() => setIsCopied(false), 2000);
+  };
+
+  const abbrAddress: any = (addr: `string`) =>
```

Comment:
please use `human-readable` function names
- suggestion: `truncateAddress`
- please also move this to a standalone, shared, lib